pkg/epinio/edit/applications.vue

```diff
@@ -100,15 +142,59 @@ export default Vue.extend<Data, any, any, any>({
         // ...changes.services
       ]);
     },
-  }
 
+    cancel() {
+      this.$router.replace(this.value.listLocation);
+    },
+
+    finish() {
+      this.$router.replace(createEpinioRoute(`c-cluster-resource-id`, {
+        cluster:  this.$store.getters['clusterId'],
+        resource: this.value.type,
+        id:       `${ this.value.meta.namespace }/${ this.value.meta.name }`
+      }));
+    },
+
+    updateSource(changes: EpinioAppSource) {
+      this.source = {};
+      const { appChart, ...cleanChanges } = changes;
+
+      this.value.configuration = this.value.configuration || {};
+
+      if (appChart) {
+        // app chart actually belongs in config, so stick it in there
+        this.set(this.value.configuration, { appchart: appChart });
+      }
+
+      if (changes.type === APPLICATION_SOURCE_TYPE.GIT_HUB) {
```

Comment:
Shouldn't this be doing the stripping of env vars if the user switches back out of a app env var based app source? If so do we need to do it in the app model?